src/io/free_functions.rs

```diff
@@ -80,9 +58,11 @@ pub fn load<R: BufRead + Seek>(r: R, format: ImageFormat) -> ImageResult<Dynamic
         #[cfg(feature = "ico")]
         image::ImageFormat::Ico => DynamicImage::from_decoder(ico::IcoDecoder::new(r)?),
         #[cfg(feature = "hdr")]
-        image::ImageFormat::Hdr => DynamicImage::from_decoder(hdr::HdrAdapter::new(BufReader::new(r))?),
+        image::ImageFormat::Hdr => DynamicImage::from_decoder(hdr::HdrAdapter::new(r)?),
+        #[cfg(feature = "openexr")]
+        image::ImageFormat::OpenExr => DynamicImage::from_decoder(openexr::ExrDecoder::read(r, None)?),
         #[cfg(feature = "pnm")]
-        image::ImageFormat::Pnm => DynamicImage::from_decoder(pnm::PnmDecoder::new(BufReader::new(r))?),
```

Comment:
Made this radical change while I was at it. In the rare case an unbuffered reader is used, these two image formats now might have worse performance, but now it's at least equal for all image formats. In the image crate, this reader is always already be buffered. I also clarified this in the documentation.

Reply:
See the changes in `io::reader.rs` below

Reply:
Seems like a reasonable change, but I'd feel better if it could go in its own quick PR.

Reply:
Reverted the change, but kept the documentation